### ISSUE TITLE
add_response_headers not being added when OAuth2 filter is used

### ISSUE THREAD

**Issue by skout23** [datawire/ambassador#3333]
**Describe the bug**
When using `add_response_headers` in a Mapping or in the `ambassador Module defaults`  those headers are not added when an OAuth2 filter is triggered.  The initial 307 redirect does not include the headers. 

**To Reproduce**
1. setup 2 `Host` endpoints 
2. Implement an OAuth2 filter https://www.getambassador.io/docs/edge-stack/latest/topics/using/filters/oauth2/
3  Associate the filter and filterpolicy to one of the Host's 
4. Add `Cache-Control: no-cache, no-store` to `add_response_headers:` in the `httpmapping:` of the `ambassador Module defaults`
5. `curl -v` to both Host end-points with the OAuth2 filter and the one without.  
6. Notice the Host without will have the cache-control header, and the filtered host will respond initially with a 307 redirect however will not have the cache-control header.

These same steps can be repeated by adding the headers in a `Mapping` and will have the same results

**Expected behavior**
adding headers in a Mapping or in the `ambassador Module defaults` should include the headers regardless of filters

**Versions (please complete the following information):**
 - Ambassador: 6.6.2
 - Kubernetes environment: EKS latest version as of reporting

**Comment by kflynn** [comment]
This is correct -- `add_response_headers` happens after authentication; it's basically the last thing in the chain, after the upstream responds. (`add_request_headers` happens a bit earlier, but still after authentication.) 

What's the use case that you're wrestling with here?
